src/js/pie.js

```diff
@@ -0,0 +1,182 @@
+        gradeNames: ["gpii.d3ViewComponent", "autoInit"],
+        model: {
+            // dataSet accepts:
+            // 1. an array of primitive values, such as numbers;
+            // 2. an array of objects. The "value" element of each object needs to containe the value for drawing each pie slice.
+            // Example: [{id: string, value: number} ... ]
+            dataSet: []
+        },
+        pieOptions: {
+            width: 300,
+            height: 300,
+            // An array of colors to fill slices generated for corresponding values of model.dataSet
+            // Or, a d3 color scale that's generated based off an array of colors
+            colors: null,
+            outerRadius: null,
+            innerRadius: null,
+            animationDuration: 750
+        },
+        styles: {
+            pie: "gpii-ca-pieChart-pie",
+            slice: "gpii-ca-pieChart-slice",
+            text: "gpii-ca-pieChart-text"
+        },
+        selectors: {
+            pie: ".gpiic-ca-pieChart-pie",
+            slice: ".gpiic-ca-pieChart-slice",
+            text: ".gpiic-ca-pieChart-text"
+        },
+        events: {
+            onPieCreated: null  // Fire when the pie is created. Ready to register D3 DOM event listeners
+        },
+        listeners: {
+            "onCreate.create": {
+                funcName: "gpii.chartAuthoring.pieChart.pie.create",
+                args: ["{that}"]
+            }
+        },
+        modelListeners: {
+            dataSet: {
+                funcName: "{that}.draw",
+                excludeSource: "init"
+            }
+        },
+        invokers: {
+            draw: {
+                funcName: "gpii.chartAuthoring.pieChart.pie.draw",
+                args: ["{that}"]
+            },
+            textTransform: {
+                funcName: "gpii.chartAuthoring.pieChart.textTransform",
+                args: ["{that}.arc", "{arguments}.0"]
+            }
+        }
+    });
+
+    gpii.chartAuthoring.pieChart.pie.draw = function (that) {
+        var svg = that.svg,
+            pie = that.pie,
+            arc = that.arc,
+            color = that.colorScale,
+            dataSet = that.model.dataSet,
+            sliceClass = that.classes.slice,
+            textClass = that.classes.text,
+            animationDuration = that.options.pieOptions.animationDuration;
+
+        var paths = svg.selectAll("path")
+            .data(pie(dataSet));
+        var texts = svg.selectAll("text")
+            .data(pie(dataSet));
+
+        // Draw pie slices
+        paths.enter()
+            .append("path")
+            .attr({
+                "fill": function(d, i) {
+                    return color(i);
+                },
+                "d": arc,
+                "class": sliceClass
+            })
+            .each(function (d) {
+                this._current = d;
+            });
+
+        paths.transition().duration(animationDuration).attrTween("d", function (a) {
```

Comment:
Part of the issue is, I think, that we're working in some cases from example functions from the D3 documentation about how to accomplish common visualization tasks. D3's conventions encourage single-character variable names. I don't love it either; I'll try & clarify what the functions are doing better.